newbank/server/NewBank.java

```diff
@@ -39,35 +40,40 @@ public synchronized CustomerID checkLogInDetails(String userName, String passwor
 
 	// commands from the NewBank customer are processed in this method
 	public synchronized String processRequest(CustomerID customer, String request) {
-		if(customers.containsKey(customer.getKey())) {
-			/* I replaced the switch cases with if statements to verify certain conditions for the input String,
-			   which I think is easier with if statements.*/
-			if (request.equals("SHOWMYACCOUNTS")) {
-				return showMyAccounts(customer);
-			}
-
-			// NEWACCOUNT request: First verifies if the request starts with NEWACCOUNT followed by a blank symbol.
-			else if (request.startsWith("NEWACCOUNT ")) {
-				/* splits the user input into a list at the first occurrence of a blank symbol.
-				Everything after the blank symbol is part of the account name. */
-				String accountName = request.split(" ", 2)[1];
-
-				// guard against too short or too long account name.
-				if (accountName.length() < 2 || accountName.length() > 20){
-					return "Please retry enter between 2 and 20 characters for your account name.";
-				}
-				// retrieve Customer object for current customer and create a new account with an opening balance of 0.
-				Customer currentCustomer = customers.get(customer.getKey());
-				currentCustomer.addAccount(new Account(accountName, 0));
-
-				// return success message containing new account name and balance.
-				return "Successfully opened the account " + "\"" + accountName + "\"" + " with an initial balance 0f 0.";
+		//moving back to switch statements as it's more readable code
+		//if-else statements are more suited for boolean values and switch statements are faster in terms of compilation
+		//reference: https://www.geeksforgeeks.org/switch-vs-else/
+		if(customers.containsKey(customer.getKey())) {List<String> input = Arrays.asList(request.split("\\s*,\\s*"));
+			switch(input.get(0)) {
+				case "1" : return showMyAccounts(customer);
+				case "2" : return createNewAccount(customer, request);
+				case "DISPLAYSELECTABLEACCOUNTS" : return displaySelectableAccounts(customer);
```

Comment:
the method displaySelectableAccounts() cannot be resolved according to IntelliJ, could not find it in any class. Did you mean showMyAccounts(), or maybe you forgot to add the method?

Reply:
theres an issue with this case, will remove it for now...